hvantk/datasets/peptideatlas_phospho_datasets.py

```
@@ -0,0 +1,418 @@
+        zip_filename = f"atlas_build_{self.build_id}.tsv.zip"
+        zip_path = os.path.join(output_dir, zip_filename)
+        tsv_path = os.path.join(output_dir, f"peptideatlas-phospho-{self.build_date}.tsv")
+
+        if os.path.exists(tsv_path) and not overwrite:
+            logger.info("Intermediate TSV already exists: %s", tsv_path)
+            return tsv_path
+
+        # Download zip if needed
+        if not os.path.exists(zip_path) or overwrite:
+            logger.info("Downloading %s -> %s", self.zip_url, zip_path)
+            urllib.request.urlretrieve(self.zip_url, zip_path)
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)